File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int res= Integer.MAX_VALUE;
        int n = nums.length;
        int high = 0;
        int low = 0;
        int sum = 0;
        int len = 0;
       
         while(high < n){
            sum += nums[high];

            while(sum >= target){
                 len = high - low + 1;
             res = Math.min(len, res);
             sum -= nums[low];
                low++;
            }
            high++;
         }
        // return res == Integer.MAX_VALUE ? 0 : res;
        if(res == Integer.MAX_VALUE){
            return 0;
        }
        return res;

    }
}

File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        i =0
        j=0
        to=0
        res =float('inf')
        while j < len(nums):
            to +=nums[j]
            while to >= target:
                res = min(res, j - i + 1)
                to -= nums[i]
                i += 1
            
            j+=1

        return 0 if res == float('inf') else res
